browser/ui/views/sidebar/sidebar_items_contents_view.cc

```diff
@@ -484,16 +486,20 @@ void SidebarItemsContentsView::OnItemPressed(const views::View* item,
   auto* controller = browser_->sidebar_controller();
   auto index = GetIndexOf(item);
   if (controller->IsActiveIndex(index)) {
-    // TODO(simonhong): This is for demo. We will have another UI for closing.
-    // De-activate active item.
-    controller->ActivateItemAt(absl::nullopt);
+    controller->ActivatePanelItem();
     return;
   }
 
   WindowOpenDisposition open_disposition = WindowOpenDisposition::CURRENT_TAB;
   if (event_utils::IsPossibleDispositionEvent(event))
     open_disposition = ui::DispositionFromEventFlags(event.flags());
 
+  const auto& item_model = controller->model()->GetAllSidebarItems()[*index];
+  if (item_model.open_in_panel) {
+    controller->ActivatePanelItem(item_model.built_in_item_type);
+    return;
+  }
```

Comment:
I think we can move this part up to line 492, as we don't use open disposition in this case.

Reply:
Moved.